docs/03_server/04_event-handler/03_advanced.md

```diff
@@ -241,110 +251,247 @@ eventHandler {
+APPROVAL_TYPE                            REMOVE                                   ENUM[NEW UPDATE REMOVE UNKNOWN]
+ENTITY_ID                                3                                        STRING              
+ENTITY_TABLE                             ISSUER                                   STRING              
+-------------------------------------------------------------------------------------------
+```
+
+### Available pending approval events
 
 Once in the `APPROVAL` table, the pending event can be cancelled, rejected or accepted by sending the following event messages to GENESIS_CLUSTER: 
 
 - EVENT_PENDING_APPROVAL_ACCEPT
 - EVENT_PENDING_APPROVAL_CANCEL
 - EVENT_PENDING_APPROVAL_REJECT
 
-All messages accept a `REASON_CODE` in their metadata.
-
-The platform ensures that users cannot approve their own events. Additional levels of control (e.g. based on user groups) can be added to the front end, to the event validate method, or can be specified in server-side configuration.
-
-To configure the allowed approvers in a server-side configuration, you need to create a new xml file with the following content. You need to add the file name to the GENESIS_CLUSTER `<config></config>` element in the site-specific version of the **genesis-processes.xml**:
-
-```xml
-<genesisCluster>
-    <preExpression>
-        <![CDATA[
-
-        ]]>
-    </preExpression>
-
-    <pendingApproval>
-        <insertPendingApproval>
-            <![CDATA[
-               true
-            ]]>
-        </insertPendingApproval>
-        <acceptPendingApproval>
-            <![CDATA[
-               true
-            ]]>
-        </acceptPendingApproval>
-        <rejectPendingApproval>
-            <![CDATA[
-               true
-            ]]>
-        </rejectPendingApproval>
-        <cancelPendingApproval>
-            <![CDATA[
-               true
-            ]]>
-        </cancelPendingApproval>
-    </pendingApproval>
-</genesisCluster>
+All messages require a valid `APPROVAL_ID` and `APPROVAL_MESSAGE` in their metadata.
+
+### Configuring allowed approvers: Basic
+
+The platform ensures that users cannot approve or reject their own events, only cancel them. Likewise, users that have not created a specific pending approval event can only accept or reject, not cancel. Additional levels of control (e.g. based on user groups) can be added to the front end, to the event `onValidate` method, or can be specified in the server-side configuration.
+
+To configure the allowed approvers using server-side configuration, you need to create a new GPAL approval file, and its name must end in **-approval.kts** (e.g. **_test_-approval.kts**). You need to add the file name to the GENESIS_CLUSTER `<script></script>` element in the site-specific version of the **genesis-processes.xml**. See the sample file below:
+
+```kotlin
+pendingApproval {
+    insert {
+        true
+    }
+
+    accept {
+        true
```

Comment:
I think we should have an example where the user has a right... otherwise people can approve each others

Reply:
ah sorry it is called out a bit later... maybe put same in this example?

Reply:
will do, and also add a rightsummarycache example.